datafusion/sql/src/expr/mod.rs

```diff
@@ -224,14 +225,26 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
             SQLExpr::Cast {
                 expr, data_type, ..
-            } => Ok(Expr::Cast(Cast::new(
-                Box::new(self.sql_expr_to_logical_expr(
-                    *expr,
-                    schema,
-                    planner_context,
-                )?),
-                self.convert_data_type(&data_type)?,
-            ))),
+            } => {
+                let dt = self.convert_data_type(&data_type)?;
+                let expr =
+                    self.sql_expr_to_logical_expr(*expr, schema, planner_context)?;
+
+                // int/floats input should be treated as seconds rather as nanoseconds
```

Comment:
```suggestion
                // numeric constants are treated as seconds (rather as nanoseconds)
                // to align with postgres / duckdb semantics. 
```

Reply:
I was worried about this change as it handles the case of `CAST (x to TIMESTAMP)` but I was worried that it would be inconsistent with automatic coercion rules for numbers. 

Then I tried some examples and I think there are no automatic coercion rules for integers to timestamps, which makes sense:

```
❯ select now() - 100000;
Error during planning: Cannot coerce arithmetic expression Timestamp(Nanosecond, Some("+00:00")) - Int64 to valid types
❯ select now() > 100000;
Error during planning: Cannot infer common argument type for comparison operation Timestamp(Nanosecond, Some("+00:00")) > Int64
```